microprofile/microprofile-rest-client-1.3/src/test/java/org/wildfly/swarm/ts/microprofile/rest/client/v13/SslSupportTest.java

```diff
@@ -0,0 +1,73 @@
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.security.KeyStore;
+
+import org.apache.http.client.fluent.Request;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.ClassLoaderAsset;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Tests new SSL support for rest client.
+ * In pom.xml there is keytool-maven-plugin that generates KS,TS.
+ * In project-defaults.yml there are server side KS,TS loaded.
+ * ShrinkWrap adds server and client side KS,TS.
+ * Programmatic client is created with loading /META-INF/client-*store.jks
+ * CDI client has settings in microprofile-config.properties thanks to configKey.
+ */
+@RunWith(Arquillian.class)
+public class SslSupportTest {
+    private static final char[] PASSWORD = "client-password".toCharArray();
+
+    @Deployment
+    public static Archive<?> deployment() {
+        return ShrinkWrap.create(WebArchive.class)
+                .addPackage(SslSupportTest.class.getPackage())
+                .addAsWebInfResource(EmptyAsset.INSTANCE, "beans.xml")
+                .addPackages(true, "org.wildfly.swarm.ts.microprofile.rest.client.v13")
+
+                .addAsManifestResource("client-cert.cer", "client-cert.cer")
+                .addAsManifestResource("client-keystore.jks", "client-keystore.jks")
+                .addAsManifestResource("client-truststore.jks", "client-truststore.jks")
+                .addAsManifestResource("server-cert.cer", "server-cert.cer")
+                .addAsManifestResource("server-keystore.jks", "server-keystore.jks")
+                .addAsManifestResource("server-truststore.jks", "server-truststore.jks")
+
+                .addAsManifestResource("META-INF/microprofile-config.properties", "microprofile-config.properties")
+                .addAsResource(new ClassLoaderAsset("project-defaults.yml"), "project-defaults.yml");
+    }
+
+    @Test
+    @RunAsClient
+    public void testCorrectClient() throws IOException {
+        String response = Request.Get("http://localhost:8080/rest/client/ssl-client").execute().returnContent().asString();
+        assertThat(response).isEqualTo("Hello from SSL client.");
+    }
+
+    @Test
+    @RunAsClient
+    public void testCdiClient() throws IOException {
+        String response = Request.Get("http://localhost:8080/rest/client/cdi-ssl-client").execute().returnContent().asString();
+        assertThat(response).isEqualTo("Hello from SSL cdi client.");
+    }
+
+    public static KeyStore getKeyStore(File keystoreFile) throws Exception {
```

Comment:
Is this method used somewhere? It doesn't seem to be used in this class.

Reply:
removed